UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common.UnitTests/Helpers/FileShareServiceCacheTest.cs

```diff
@@ -0,0 +1,174 @@
+            A.CallTo(() => fakeAzureTableStorageClient.RetrieveAsync<FssResponseCache>(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(GetResponseCache());
+            A.CallTo(() => fakeAzureBlobStorageClient.GetCloudBlockBlob(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(new CloudBlockBlob(new System.Uri("http://tempuri.org/blob")));
+            A.CallTo(() => fakeFileSystemHelper.DownloadToFileAsync(A<CloudBlockBlob>.Ignored, A<string>.Ignored));
+
+            var response = await fileShareServiceCache.GetNonCacheProductDataForFss(GetProductdetails(), GetSearchBatchResponse(), exchangeSetRootPath, GetScsResponseQueueMessage(), null, CancellationToken.None);
+
+            Assert.IsNotNull(response);
+            Assert.IsInstanceOf(typeof(List<Products>), response);
+        }
+
+        [Test]
+        public async Task WhenGetNonCacheProductDataForFssReturnsProductListNotFound()
+        {
+            var cachingResponse = new FssResponseCache() { };
+            A.CallTo(() => fakeAzureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns(GetStorageAccountConnectionStringAndContainerName().Item1);
+            A.CallTo(() => fakeAzureTableStorageClient.RetrieveAsync<FssResponseCache>(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(cachingResponse);
+
+            var response = await fileShareServiceCache.GetNonCacheProductDataForFss(GetProductdetails(), null, string.Empty, GetScsResponseQueueMessage(), null, CancellationToken.None);
+
+            Assert.IsNotNull(response);
+            Assert.IsInstanceOf(typeof(List<Products>), response);
+        }
+
+        [Test]
+        public void WhenGetNonCacheProductDataForFssReturnsCancellation()
+        {
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+            cancellationTokenSource.Cancel();
+            CancellationToken cancellationToken = cancellationTokenSource.Token;
+            
+            Assert.ThrowsAsync<OperationCanceledException>(async () => await fileShareServiceCache.GetNonCacheProductDataForFss(GetProductdetails(), GetSearchBatchResponse(), string.Empty, GetScsResponseQueueMessage(), cancellationTokenSource, cancellationToken));
+        }       
+
+        [Test]
+        public async Task WhenInsertOrMergeFssCacheDetailReturnsTrue()
+        {            
+            A.CallTo(() => fakeAzureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns(GetStorageAccountConnectionStringAndContainerName().Item1);
+            A.CallTo(() => fakeAzureTableStorageClient.InsertOrMergeAsync(A<TableEntity>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(GetResponseCache());
+            await fileShareServiceCache.InsertOrMergeFssCacheDetail(GetResponseCache());
+
+            Assert.IsNotNull(true);
+        }
+    }
+}
```

Comment:
Kindly change test method names to

WhenGetNonCacheProductDataForFssIsCalled_ThenReturnProductNotFound

WhenGetNonCacheProductDataForFssIsCalled_ThenReturnProductFound

WhenCancellationRequestedInGetNonCacheProductDataForFss_ThenThrowOperationCanceledException

WhenInsertOrMergeFssCacheDetail_ThenReturnTrue